src/_pytest/logging.py

```diff
@@ -513,6 +509,8 @@ def pytest_sessionfinish(self):
 
     @pytest.hookimpl(hookwrapper=True, tryfirst=True)
     def pytest_sessionstart(self):
+        # This has to be called before the first log message is logged,
```

Comment:
im wondering - shouldn't we do this as early as `pytest_configure`

Reply:
Yes, maybe. We can call it once the terminalwrapper and capturemanager  plugins are configured (see code of _setup_cli_logging).

Reply:
I've just stumbled upon this code in LoggingPlugin.__init__ (which is called in pytest_configure):

```py
            # sanity check: terminal reporter should not have been loaded at this point
            assert self._config.pluginmanager.get_plugin("terminalreporter") is None
```

So I guess that both capturemanager and terminalreporter are not available in pytest_configure

Reply:
@thisch i believe a `@hookimpl(trylast=True)` can sort that

Reply:
it may be more tricky :/ i didnt account for setuptools entrypoints